### PR TITLE
Viafirma

### PR DESCRIPTION
**Pull request by anfuca**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/PDF4me Connect/apiDefinition.swagger.json [hunk 1]
```diff
@@ -2461,6 +2461,315 @@
           }
```
Comment (sriyen-msft) [inline certified-connectors/PDF4me Connect/apiDefinition.swagger.json:2461]: @anfuca Your PR contains other connector file changes which is not allowed. Please remove them from your PR. Please raise one PR for each connector.

File: certified-connectors/Viafirma/Readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Viafirma Connector
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/Readme.md:1]: @anfuca Please follow the formatting of [the readme.md template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/readme.md) and provide the required information.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "groupCode": {
+                  "type": "string",
+                  "description": "groupCode",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:55]: @anfuca All the description values of properties are expected to be free from grammatical and spell errors. Please provide valid description values.
Docs - [Define operation and parameter summaries and descriptions](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#define-operation-and-parameter-summaries-and-descriptions)

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+                  "title": "",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:56]: @anfuca Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+                    "value-title": "code"
+                  }
+                },
+                "workflow": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "type",
+                      "title": "",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:70]: @anfuca Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+                }
+              },
+              "default": {
+                "groupCode": "your_group_here",
+                "workflow": {
+                  "type": "WEB"
+                },
+                "notification": {
+                  "text": "TEST Firma web",
+                  "detail": "notificado v\u00eda email",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:141]: @anfuca Unicodes - The connector's swagger still have unicodes which may not be supported in all the Power Automate page, Can you please replace all the unicodes with respective characters in the English language.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+                "notification"
+              ]
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "200",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:166]: @anfuca Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+                "groupCode"
+              ]
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "200",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:265]: @anfuca Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+          },
+          "default": {
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:267]: @anfuca Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+            "description": "default"
+          }
+        },
+        "x-ms-visibility": "important"
+      }
+    },
+    "/documents/api/v3/system/alive": {},
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:274]: @anfuca Empty actions can be removed as they would not add any value to your connector.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+    "/documents/api/v3//template/list/mine": {},
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:275]: @anfuca Empty actions can be removed as they would not add any value to your connector.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,341 @@
...
+    "/documents/api/v3/template/list/mine": {
+      "get": {
+        "summary": "Get user's templates",
+        "description": "List all the user's templates",
+        "operationId": "GetUserTemplates",
+        "parameters": [],
+        "responses": {
+          "200": {
+            "description": "200",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:285]: @anfuca Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,341 @@
...
+          },
+          "default": {
+            "description": "default"
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:288]: @anfuca Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,341 @@
...
+          }
+        },
+        "x-ms-visibility": "internal"
+      }
+    },
+    "/documents/api/v3/users/whoiam/": {
+      "get": {
+        "responses": {
+          "default": {
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:297]: @anfuca Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger.

File: certified-connectors/Viafirma/apiProperties.json [hunk 15]
```diff
@@ -0,0 +1,35 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "oauth2",
+          "clientId": "com.viafirma.doc.app",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiProperties.json:8]: @anfuca Please remove the client id value as per Microsoft's security policy. You can replace it with dummy text.
`"clientId": "[[DUMMY]]"`

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by anfuca** [comment]
@sriyen-msft I've already fixed all the stuff. Also, i've tried (with no luck) to remove the initial commit of this pull request. 
The thing is that now all the commits have disappeared and the PR is closed. 
How can I reopened the PR and commit the new changes?
Thanks in advance

**Review by anfuca — COMMENTED** [review]
Ok I'll undo the first commit

File: certified-connectors/Viafirma/apiProperties.json (hunk 15 above)
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiProperties.json:8]: @anfuca Please remove the client id value as per Microsoft's security policy. You can replace it with dummy text.
`"clientId": "[[DUMMY]]"`

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 16]
```diff
@@ -0,0 +1,383 @@
...
+                },
+                "rol": {
+                  "type": "string",
+                  "description": "User rol"
+                }
+              }
+            }
+          }
+        },
+        "summary": "WhoIAm",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:343]: @anfuca All the summary and description values of properties are expected to be free from grammatical and spell errors. Please provide valid description values.
Docs - [Define operation and parameter summaries and descriptions](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#define-operation-and-parameter-summaries-and-descriptions)

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 17]
```diff
@@ -0,0 +1,383 @@
...
+                    "value-title": "groups-key-item-output"
+                  }
+                },
+                "workflow": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "type",
+                      "title": "",
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:217]: @anfuca Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/Viafirma/apiDefinition.swagger.json [hunk 18]
```diff
@@ -0,0 +1,383 @@
...
+                    "type"
+                  ]
+                },
+                "document": {
+                  "type": "object",
+                  "properties": {
+                    "templateCode": {
+                      "type": "string",
+                      "description": "templateCode",
+                      "title": ""
```
Comment (sriyen-msft) [inline certified-connectors/Viafirma/apiDefinition.swagger.json:233]: @anfuca Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/PDF4me Connect/apiDefinition.swagger.json [hunk 19]
```diff
@@ -2461,6 +2461,315 @@
...
         ]
       }
+    },
```
Comment (sriyen-msft) [inline certified-connectors/PDF4me Connect/apiDefinition.swagger.json:2464]: @anfuca Please remove this file from your PR.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/Viafirma/apiDefinition.swagger.json (hunk 3 above)
Comment (sriyen-msft) [reply certified-connectors/Viafirma/apiDefinition.swagger.json:55]: @anfuca It is not updated as requested, Please let us know the reason for resolving this comment?

File: certified-connectors/PDF4me Connect/apiDefinition.swagger.json [hunk 20]
```diff
@@ -1,2496 +0,0 @@
-{
```
Comment (sriyen-msft) [inline certified-connectors/PDF4me Connect/apiDefinition.swagger.json:1]: @anfuca You need to remove other connector files from your PR. But you have removed it from the repo and if your PR gets approved and merged, the other connector swagger would be removed from our repo. Can you please sync your repo with our dev repo and update your PR with only your connector files.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**